### PR TITLE
add OpenMP support to sunmd5

### PR DESCRIPTION
**Pull request by zzlei**
remove some redundant comments BTW

### CODE REVIEW

File: src/sunmd5_fmt_plug.c [hunk 1]
```diff
@@ -159,20 +159,11 @@ static struct fmt_tests tests[] = {
...
-unsigned char input_buf_big[25][BLK_CNT*MD5_CBLOCK] __attribute__ ((aligned(16)));
-unsigned char out_buf[BLK_CNT*MD5_DIGEST_LENGTH]    __attribute__ ((aligned(16)));
-*/
-#else
-static unsigned char *input_buf;
-static unsigned char *out_buf;
-static unsigned char (*input_buf_big)[BLK_CNT*MD5_CBLOCK];
+#ifdef _OPENMP
+#pragma omp threadprivate(out_buf, input_buf, input_buf_big)
 #endif
```
Comment (zzlei) [inline src/sunmd5_fmt_plug.c]: I opt to use static allocation for those arrays, to make OpenMP code clearer. This might need discussion.

**Comment by zzlei** [comment]
Performance comparison (on my laptop):

Non-OpenMP
----
Benchmarking: SunMD5 [MD5 128/128 AVX 4x3]... DONE
Speed for cost 1 (iteration count) of 5000
Raw:	605 c/s real, 610 c/s virtual

OpenMP
----
Will run 4 OpenMP threads
Benchmarking: SunMD5 [MD5 128/128 AVX 4x3]... (4xOMP) DONE
Speed for cost 1 (iteration count) of 5000
Raw:	1269 c/s real, 326 c/s virtual

File: src/sunmd5_fmt_plug.c [hunk 2]
```diff
@@ -553,319 +539,314 @@ static int crypt_all(int *pcount, struct db_salt *salt)
...
-	 */
-
-	/* this is the 'first' sprintf(roundascii,"%d",round);  The rest are at the bottom of the loop */
-	strcpy(roundascii, "0");
-	roundasciilen=1;
-
-	for (round = 0; round < maxrounds; round++) {
+#ifdef _OPENMP
+#pragma omp parallel for default(none) private(idx) copyin(input_buf_big) \
+	shared(saved_salt, data, constant_phrase, ngroups, group_sz)
```
Comment (zzlei) [inline src/sunmd5_fmt_plug.c]: Got a problem here. GCC won't allow me to mark `constant_phrase` as shared, as it's already a constant variable. But icc wants to keep it anyway. Now the situation is: keeping it, gcc fails to compile; dropping it, icc fails to compile. Damn.

Comment (magnumripper) [inline src/sunmd5_fmt_plug.c]: Lol. Well you could always use `#if __INTEL_COMPILER`

**Comment by magnumripper** [comment]
Is that benchmark using 2 real cores, 4 threads?

**Comment by zzlei** [comment]
> Is that benchmark using 2 real cores, 4 threads?

Yes.

BTW, here's the result on well:

**Non-OpenMP**
Will run 8 OpenMP threads
Benchmarking: SunMD5 [MD5 256/256 AVX2 8x3]... (8xOMP) DONE
Speed for cost 1 (iteration count) of 5000
Raw:	5485 c/s real, 683 c/s virtual

**OpenMP**
Benchmarking: SunMD5 [MD5 256/256 AVX2 8x3]... DONE
Speed for cost 1 (iteration count) of 5000
Raw:	1384 c/s real, 1371 c/s virtual

**Pull request merged**